File: 128_longest_consecutive_sequence/longest_consecutive_sequence.go
package main

import "fmt"

type Solution struct{}

func (s *Solution) longestConsecutiveSequence(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	numSet := make(map[int]bool)
	for _, num := range nums {
		numSet[num] = true
	}

	longest := 0
	for num := range numSet {
		if !numSet[num-1] {
			length := 1
			for numSet[num+length] {
				length++
			}
			if length > longest {
				longest = length
			}
		}
	}
	return longest
}

func main() {
	nums := []int{100, 4, 200, 1, 3, 2}
	s := Solution{}
	fmt.Println(s.longestConsecutiveSequence(nums))
}

File: 128_longest_consecutive_sequence/longest_consecutive_sequence.py
from typing import List

class Solution:
    def longest_consecutive_sequence(self, nums: List[int]) -> int:
        if not nums:
            return 0

        num_set = set(nums)
        longest = 0

        for num in num_set:
            if num - 1 not in num_set:
                length = 1
                while num + length in num_set:
                    length += 1
                longest = max(longest, length)

        return longest


if __name__ == "__main__":
    nums = [100, 4, 200, 1, 3, 2]
    solution = Solution()
    print(solution.longest_consecutive_sequence(nums))
